draft-ietf-tls-tls13.md

```diff
@@ -1805,6 +1806,18 @@ this specification are:
 
 - key_share (see {{key-share}})
 
+In addition, in its updated ClientHello, the client SHOULD NOT offer
+any pre-shared keys associated with a hash other than that of the selected
+cipher suite. This allows the client to avoid having to compute partial
+hash transcripts for multiple hashes in the second ClientHello.
+A client which receives a cipher suite that was not offered MUST abort the handshake.
+Servers MUST ensure that they negotiate the same cipher suite when
+receiving a conformant updated ClientHello (if the server selects the
+cipher suite first, then this will happen automatically). Upon
```

Comment:
"selects the cipher suite first" seems a little informal; maybe "this happens automatically if the server selects a negotiated ciphersuite as the first step of processing a received ClientHello".  Though, hmm, aren't we supposed to look at some extensions before we know what ciphers we can use?